runtime_test.go

```diff
@@ -122,11 +128,75 @@ func (s *runtimeSuite) TestRuntime(c *C) {
 	}
 	ok, val = failpoint.Eval("runtime-test-6")
 	c.Assert(ok, IsFalse)
+	c.Assert(val, IsNil)
 
 	ok, val = failpoint.Eval("failpoint-env1")
 	c.Assert(ok, IsTrue)
 	c.Assert(val.(int), Equals, 10)
 	ok, val = failpoint.Eval("failpoint-env2")
 	c.Assert(ok, IsTrue)
 	c.Assert(val.(bool), IsTrue)
+
+	// Tests for sleep
+	ch = make(chan struct{})
+	go func() {
+		time.Sleep(time.Second)
+		err := failpoint.Disable("gofail/test-sleep")
+		c.Assert(err, IsNil)
+		close(ch)
+	}()
+	err = failpoint.Enable("gofail/test-sleep", "sleep(100)")
+	c.Assert(err, IsNil)
+	start = time.Now()
+	ok, v = failpoint.Eval("gofail/test-sleep")
+	c.Assert(ok, IsFalse)
+	c.Assert(v, IsNil)
+	c.Assert(time.Since(start), GreaterEqual, 90*time.Millisecond, Commentf("not sleep"))
+	<-ch
+
+	// Tests for sleep duration
+	ch = make(chan struct{})
+	go func() {
+		time.Sleep(time.Second)
+		err := failpoint.Disable("gofail/test-sleep2")
+		c.Assert(err, IsNil)
+		close(ch)
+	}()
+	err = failpoint.Enable("gofail/test-sleep2", `sleep("100ms")`)
+	c.Assert(err, IsNil)
+	start = time.Now()
+	ok, v = failpoint.Eval("gofail/test-sleep2")
+	c.Assert(ok, IsFalse)
+	c.Assert(v, IsNil)
+	c.Assert(time.Since(start), GreaterEqual, 90*time.Millisecond, Commentf("not sleep"))
+	<-ch
+
+	// Tests for print
+	oldStdio := os.Stdout
+	r, w, err := os.Pipe()
+	c.Assert(err, IsNil)
+	os.Stdout = w
+	err = failpoint.Enable("test-print", `print`)
+	ok, val = failpoint.Eval("test-print")
+	c.Assert(ok, IsFalse)
+	c.Assert(val, IsNil)
+	outC := make(chan string)
+	// copy the output in a separate goroutine so printing can't block indefinitely
+	go func() {
+		var buf bytes.Buffer
+		io.Copy(&buf, r)
```

Comment:
`io.Copy` returns an error, should need to handle this?

And perhaps you could just use `ioutils.ReadAll`